src/components/AddProductFiltersModal.vue

```diff
@@ -70,76 +76,83 @@ import {
 
 const props = defineProps(["label", "facetToSelect", "searchfield", "type"]);
 const store = useStore();
 
 const appliedFilters = computed(() => store.getters["util/getAppliedFilters"]);
+const getFacetOptions = computed(() => store.getters["util/getFacetOptions"]);
 
-onMounted(() => {
+onMounted(async() => {
+  isLoading.value = true;
+  await store.dispatch("util/fetchProductFilters", { facetToSelect: props.facetToSelect, searchfield: props.searchfield })
+  facetOptions.value = getFacetOptions.value(props.searchfield);
+  availableOptions.value = JSON.parse(JSON.stringify(facetOptions.value))
+  getFilters();
   selectedValues.value = JSON.parse(JSON.stringify(appliedFilters.value[props.type][props.searchfield]))
+  isLoading.value = false;
 })
 
 function closeModal() {
   modalController.dismiss({ dismissed: true });
 }
 
 function search() {
-  getFilters();
-}
-
-async function getFilters(vSize?: any, vIndex?: any) {
-  const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
-  const viewIndex = vIndex ? vIndex : 0;
-  
-  const payload = {
-    facetToSelect: props.facetToSelect,
-    docType: 'PRODUCT',
-    coreName: 'enterpriseSearch',
-    searchfield: props.searchfield,
-    jsonQuery: '{"query":"*:*","filter":["docType:PRODUCT"]}',
-    noConditionFind: 'N',
-    limit: viewSize,
-    q: queryString.value,
-    term: queryString.value,
-    offset: viewSize * viewIndex,
-  }
-
-  const resp = await UtilService.fetchFacets(payload);
-  if(!hasError(resp)) {
-    const results = resp.data.facetResponse.response
-    facetOptions.value = viewIndex === 0 ? results : [...facetOptions.value , ...results];
-    isScrollable.value = (facetOptions.value.length % process.env.VUE_APP_VIEW_SIZE) === 0;
+  isScrollable.value = true;
+  currentPage.value = 0;
+  filteredOptions.value = []
 
+  if(queryString.value.trim()) {
+    availableOptions.value = facetOptions.value.filter((option: any) => option.label.toLowerCase().includes(queryString.value.trim().toLowerCase()))
   } else {
-    facetOptions.value = [];
-    isScrollable.value = false;
+    availableOptions.value = JSON.parse(JSON.stringify(facetOptions.value))
   }
+  getFilters();
 }
 
 async function loadMoreFilters(event: any){
-  getFilters(
-    undefined,
-    Math.ceil(facetOptions.value.length / process.env.VUE_APP_VIEW_SIZE).toString() 
-  ).then(() => {
+  // Added this check here as if added on infinite-scroll component the Loading content does not gets displayed
+  if(!(isScrollingEnabled.value && isScrollable.value)) {
+    await event.target.complete();
+  }
+
+  getFilters().then(() => {
     event.target.complete();
   })
 }
 
+async function getFilters() {
+  const nextPageItems = availableOptions.value.slice(currentPage.value * pageSize, (currentPage.value + 1) * pageSize);
+  filteredOptions.value = filteredOptions.value.concat(nextPageItems);
+  currentPage.value += 1;
+
+  if(filteredOptions.value.length >= facetOptions.value.length) {
```

Comment:
We should disable scroll when the filteredOptions becomes >= availableOptions, as if total options are 20, but after searching we got 15 options, so in that case the scroll will not become inactive and thus this function will run always on scroll